src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
```

Comment:
Ensure that the `List` interface is correctly defined or imported in your project. This is crucial for the `ArrayList` implementation to work as expected.